homeassistant/components/light/template.py

```diff
@@ -0,0 +1,242 @@
+                entity_ids)
+        )
+
+    if not lights:
+        _LOGGER.error("No lights added")
+        return False
+
+    async_add_devices(lights, True)
+    return True
+
+
+class LightTemplate(Light):
+    """Representation of a templated Light, including dimmable."""
+
+    def __init__(self, hass, device_id, friendly_name, state_template,
+                 on_action, off_action, level_action, level_template,
+                 entity_ids):
+        """Initialize the light."""
+        self.hass = hass
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, device_id, hass=hass)
+        self._name = friendly_name
+        self._template = state_template
+        self._on_script = Script(hass, on_action)
+        self._off_script = Script(hass, off_action)
+        self._level_script = Script(hass, level_action)
+        self._level_template = level_template
+
+        self._state = False
+        self._brightness = None
+        self._entities = entity_ids
+        self._entities.append(self.entity_id)
+
+        if self._template is not None:
+            self._template.hass = self.hass
+        if self._level_template is not None:
+            self._level_template.hass = self.hass
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._level_script is not None:
+            return SUPPORT_BRIGHTNESS
+
+        return 0
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        state = yield from async_get_last_state(self.hass, self.entity_id)
+        if state:
+            self._state = state.state == STATE_ON
+
+        @callback
+        def template_light_state_listener(entity, old_state, new_state):
+            """Handle target device state changes."""
+            # Determine need to evaluate template before returning.
+            yield_for_render = (self._template is not None or
+                                self._level_template is not None)
+            self.hass.async_add_job(
+                self.async_update_ha_state(yield_for_render))
+
+        @callback
+        def template_light_startup(event):
+            """Update template on startup."""
+            async_track_state_change(
+                self.hass, self._entities, template_light_state_listener)
+
+            # Determine need to evaluate template before returning.
+            yield_for_render = (self._template is not None or
+                                self._level_template is not None)
+            self.hass.async_add_job(
```

Comment:
I think here we should just run it with `async_update_ha_state(True)`. This one is only run once, so it's not a huge deal. The big concern was the updates in the service methods. Sorry for the miscommunication.